src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -391,8 +409,10 @@ template <class T> void getEscapedString(const SchemaField& field, const nlohman
 // Convert [{type: "FieldType", value: "value"}, ..] to string that can be used in collection range end
 seastar::future<nlohmann::json> HTTPProxy::_handleGetKeyString(nlohmann::json&& request) {
     String output;
-    if (!request.contains("fields"))
+    if (!request.contains("fields")) {
         return JsonResponse(Statuses::S400_Bad_Request("Invalid json"));
+        _deserializationErrors++;
```

Comment:
this line doesn't execute. i'm surprised this compiles actually - gcc warning should have errored out the build.

Reply:
Yes. Fixed.